service/system_information.go

```diff
@@ -0,0 +1,331 @@
+		}
+	}
+
+	var allNets []api.Network
+	s.AvailableUplinkInterfaces, s.AvailableCephInterfaces, allNets, err = lxd.GetNetworkInterfaces(ctx, s.ClusterName, s.ClusterAddress, s.AuthSecret)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get network interfaces on %q: %w", s.ClusterName, err)
+	}
+
+	for _, network := range allNets {
+		if network.Name == DefaultFANNetwork {
+			s.existingFanNetwork = &network
+			continue
+		}
+
+		if network.Name == DefaultOVNNetwork {
+			s.existingOVNNetwork = &network
+			continue
+		}
+
+		if network.Name == DefaultUplinkNetwork {
+			s.existingUplinkNetwork = &network
+			continue
+		}
+	}
+
+	pools, err := lxd.GetStoragePools(ctx, s.ClusterName, s.ClusterAddress, s.AuthSecret)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get storage pools on %q: %w", s.ClusterName, err)
+	}
+
+	pool, ok := pools[DefaultZFSPool]
+	if ok {
+		poolCopy := pool
+		s.existingLocalPool = &poolCopy
+	}
+
+	pool, ok = pools[DefaultCephPool]
+	if ok {
+		poolCopy := pool
+		s.existingRemotePool = &poolCopy
+	}
+
+	pool, ok = pools[DefaultCephFSPool]
+	if ok {
+		poolCopy := pool
+		s.existingRemoteFSPool = &poolCopy
+	}
+
+	// Last one is configs
+
+	if len(s.ExistingServices[types.MicroCeph]) > 0 {
+		microceph := sh.Services[types.MicroCeph].(*CephService)
+
+		if localSystem {
+			s.CephConfig, err = microceph.ClusterConfig(ctx, "", "")
+		} else {
+			s.CephConfig, err = microceph.ClusterConfig(ctx, s.ClusterAddress, s.AuthSecret)
+		}
+
+		if err != nil && err.Error() != "Daemon not yet initialized" {
+			return nil, fmt.Errorf("Failed to get ceph configuration on %q: %w", s.ClusterName, err)
+		}
+	}
+
+	if localSystem {
+		s.LXDLocalConfig, s.LXDConfig, err = lxd.GetConfig(ctx, s.ServiceClustered(types.LXD), s.ClusterName, "", "")
+	} else {
+		s.LXDLocalConfig, s.LXDConfig, err = lxd.GetConfig(ctx, s.ServiceClustered(types.LXD), s.ClusterName, s.ClusterAddress, s.AuthSecret)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get LXD configuration on %q: %w", s.ClusterName, err)
+	}
+
+	return s, nil
+}
+
+// GetExistingClusters checks against the services reachable by the specified ServerInfo,
+// and returns a map of cluster members for each service supported by the Handler.
+// If a service is not clustered, its map will be nil.
+func (sh *Handler) GetExistingClusters(ctx context.Context, connectInfo mdns.ServerInfo) (map[types.ServiceType]map[string]string, error) {
+	localSystem := sh.Name == connectInfo.Name
+	var err error
+	existingServices := map[types.ServiceType]map[string]string{}
+	for service := range sh.Services {
+		var existingCluster map[string]string
+		if localSystem {
+			existingCluster, err = sh.Services[service].ClusterMembers(ctx)
+		} else {
+			existingCluster, err = sh.Services[service].RemoteClusterMembers(ctx, connectInfo.AuthSecret, connectInfo.Address)
+		}
+
+		if err != nil && err.Error() != "Daemon not yet initialized" && err.Error() != "Server is not clustered" {
```

Comment:
The microcluster half of this error check can use `api.StatusErrorCheck` to check for a `503` once the dependencies are updated, but I'm not sure what to do for the LXD side. Since these are remote requests, it's not ideal to send a whole other request to `GetServer` just to check if the system is clustered, so is there a nice way to detect this particular error from LXD?

@tomponline got any input here?

Reply:
Try using Go client's IsClustered function.